include/openthread/platform/uart.h

```diff
@@ -83,6 +83,17 @@ otError otPlatUartDisable(void);
  */
 otError otPlatUartSend(const uint8_t *aBuf, uint16_t aBufLength);
 
+/**
+ * Flush the outgoing transmit buffer and wait for the data to be sent.
+ * This is called when the CLI UART interface has a full buffer but still
+ * wishes to send more data.
+ *
+ * @retval OT_ERROR_NONE    Flush succeeded, we can proceed to write more
+ *                          data to the buffer.
+ * @retval !OT_ERROR_NONE   Flush failed.  Following data will be truncated.
```

Comment:
How about documenting `OT_ERROR_NOT_IMPLEMENTED` and `OT_ERROR_INVALID_STATE`.

Reply:
I guess I was leaving up to the driver writer to use whatever `otError` enum they felt was appropriate. :-)